src/main/java/com/squareup/javapoet/JavaFile.java

```diff
@@ -71,14 +59,17 @@ private JavaFile(Builder builder) {
   }
 
   public void writeTo(Appendable out) throws IOException {
+    StringBuilder temp = new StringBuilder();
     // First pass: emit the entire class, just to collect the types we'll need to import.
-    CodeWriter importsCollector = new CodeWriter(NULL_APPENDABLE, indent, staticImports);
+    CodeWriter importsCollector = new CodeWriter(temp, indent, staticImports);
```

Comment:
The content of `temp` is not used - but the allocated memory will be re-used be the second pass.